src/main/java/com/unascribed/fabrication/loaders/LoaderBlockLogo.java

```diff
@@ -113,23 +118,18 @@ public void load(Path configDir, QDIni config, boolean loadError) {
 				int swapped = colorInt&0x0000FF00;
 				swapped |= (colorInt&0x00FF0000) >> 16;
 				swapped |= (colorInt&0x000000FF) << 16;
-				List<Resolvable<Block>> blocks = Lists.newArrayList();
+				List<BlockState> blocks = Lists.newArrayList();
```

Comment:
Have you tested this with blocks added by other mods? The usage of `Resolvable` here is because Loaders get invoked early during game initialization (before mod initializers, in fact, as they are managed by Fabrication's Mixin plugin), and have to reference registries at an arm's length.

The return value from the Suppliers in colorToState are cached by the block logo renderer, so it would be better here to replace List<Resolvable<Block>> with something like List<String> and do the argument parsing in the supplier.

Reply:
Great point, I admit I didn't even think of that. Just tested with Tech Reborn blocks and it seems to work fine, however I fear this is just a recipe for an initialization order fiasco. I'll replace with a list of Strings for late name resolution.